cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java

```diff
@@ -0,0 +1,170 @@
+        targetConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION));
+
+        deserializationFacade =
+                GlueSchemaRegistryDeserializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(sourceConfigs)
+                        .build();
+
+        serializationFacade   =
+                GlueSchemaRegistrySerializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(targetConfigs)
+                        .build();
+
+        serializer   = new GlueSchemaRegistrySerializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(targetConfigs));
+
+        deserializer = new GlueSchemaRegistryDeserializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(sourceConfigs));
+
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        byte[] bytes = (byte[]) value;
+        if (value == null) return new byte[0];
+
+        try {
+            byte[] deserializedBytes = deserializer.getData(bytes);
+            Schema deserializedSchema = deserializer.getSchema(bytes);
+
+            byte[] encodedByte = serializer.encode(null, deserializedSchema, deserializedBytes);
+            com.amazonaws.services.schemaregistry.common.Schema returnedSchema = getSchema(bytes);
+            UUID uuid = registerSchema(returnedSchema);
+
+            return encodedByte;
+
+        } catch (SerializationException | AWSSchemaRegistryException e){
+            throw new DataException("Converting Kafka Connect data to byte[] failed due to serialization/deserialization error: ", e);
+        }
+
+    }
+
+    /**
+     * This method is not intended to be used for the CrossRegionReplicationMM2Converter given that MM2 is a source connector
+     *
+     */
+    @Override
+    public SchemaAndValue toConnectData(String topic, byte[] value) {
+
+        try {
+            throw new IllegalAccessException("This method is not intended to be used here");
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    /**
+     * Retrieve schema from source region GSR using schema header of the serialized messages
+     * @param data serialized message obtained from MM2
+     * @return schema
+     */
+    public Schema getSchema(byte[] data){
+        if (data == null) {
+            throw new NullPointerException("Empty Data");
+        }
+            return deserializationFacade.getSchema(data);
+    }
+
+    /**
+     * Register schema in the target region GSR
+     * @param schema schema obtained from the source region GSR
+     * @return schema version ID of the registered schema
+     */
+    public  UUID registerSchema(com.amazonaws.services.schemaregistry.common.Schema schema){
```

Comment:
How is the registry chosen in the target region?  Also, how is the schema named?

Reply:
This function is longer used